pkg/cli/tasks.go

```diff
@@ -0,0 +1,301 @@
+		if err != nil {
+			return fmt.Errorf("failed to get project %s: %w", l.ProjectID, err)
+		}
+		l.AssistantID = project.AssistantID
+		debugPrint("Found assistant ID %s for project %s", l.AssistantID, l.ProjectID)
+	}
+
+	var (
+		allTasks types.TaskList
+	)
+
+	// 1. First fetch projects to know what we have access to
+	debugPrint("Fetching projects...")
+	startProjects := time.Now()
+
+	projects, err := l.root.Client.ListProjects(ctx, apiclient.ListProjectsOptions{
+		All: l.All,
+	})
+	if err != nil {
+		debugPrint("Error listing projects: %v", err)
+		return fmt.Errorf("failed to list projects: %w", err)
+	}
+	debugPrint("Fetched %d projects in %.2f seconds", len(projects.Items), time.Since(startProjects).Seconds())
+
+	// Build lookup maps for quick access
+	projectMap := make(map[string]types.Project)
+	for _, project := range projects.Items {
+		projectMap[project.ID] = project
+	}
+
+	// If specific project is requested, only get tasks for that project
+	if l.ProjectID != "" && l.AssistantID != "" {
+		debugPrint("Using specific project %s with assistant %s", l.ProjectID, l.AssistantID)
+		tasks, err := l.root.Client.ListProjectTasks(ctx, l.AssistantID, l.ProjectID)
+		if err != nil {
+			return fmt.Errorf("failed to list tasks for project %s: %w", l.ProjectID, err)
+		}
+		allTasks = tasks
+	} else {
+		// Otherwise aggregate tasks from all accessible projects
+		taskChan := make(chan types.Task)
+		errChan := make(chan error, len(projects.Items))
+		doneChan := make(chan struct{})
+
+		// Keep track of how many goroutines we've started
+		var wg sync.WaitGroup
+
+		// Start a collector to aggregate all tasks
+		go func() {
+			for task := range taskChan {
+				allTasks.Items = append(allTasks.Items, task)
+			}
+			close(doneChan)
+		}()
+
+		// Browser-like connection limit
+		// Use a semaphore to limit concurrent API calls
+		sem := make(chan struct{}, 8)
+
+		// Filter out projects without assistantID to avoid unnecessary API calls
+		validProjects := 0
+		for _, project := range projects.Items {
+			if project.AssistantID == "" {
+				debugPrint("Skipping project %s (no assistant ID)", project.ID)
+				continue
+			}
+			validProjects++
+
+			wg.Add(1)
+			go func(p types.Project) {
+				defer wg.Done()
+
+				// Acquire semaphore slot
+				sem <- struct{}{}
+				defer func() { <-sem }() // Release slot when done
+
+				startTask := time.Now()
+				debugPrint("Fetching tasks for project %s (assistant: %s)", p.ID, p.AssistantID)
+
+				tasks, err := l.root.Client.ListProjectTasks(ctx, p.AssistantID, p.ID)
+				if err != nil {
+					debugPrint("Error listing tasks for project %s: %v", p.ID, err)
+					errChan <- err
+					return
+				}
+
+				for _, task := range tasks.Items {
+					taskChan <- task
+				}
+
+				debugPrint("Fetched %d tasks for project %s in %.2f seconds",
+					len(tasks.Items), p.ID, time.Since(startTask).Seconds())
+			}(project)
+		}
+
+		// Wait for all fetches to complete, then close channels
+		go func() {
```

Comment:
nit: this doesn't have to be a goroutine.

Reply:
removed.